### PR TITLE
Segregate juju_controller and clusterd

### PR DESCRIPTION
**Pull request by hemanthnakkina**
Currently clusterd, tls apps are running on juju_controller nodes.
Deploy clusterd, tls apps on different nodes.

Create new role/tag 'infra'
Create new model openstack-infra to deploy nodes with tag infra
Deploy Clusterd and tls-operator application on infra model machines

### CODE REVIEW

File: sunbeam-python/sunbeam/provider/maas/deployment.py [hunk 1]
```diff
@@ -142,6 +146,11 @@ def infrastructure_model(self) -> str:
         """Return the infrastructure model name."""
         return "openstack-machines"
 
+    @property
+    def cluster_model(self) -> str:
```
Comment (gboutry) [inline sunbeam-python/sunbeam/provider/maas/deployment.py:150]: We need to think about the model names, they don't make sense as a whole.

We have:
- openstack
- openstack-machines
- sunbeam-cluster

While we already have a discussion about renaming `openstack`, I think we should think about the `sunbeam-cluster`, maybe renaming it to `openstack-infra` or something related? What do you think?


Of course, if the name becomes openstack-infra, we'll need to rename the property `infrastructure_model` to something else

**Review by gboutry — COMMENTED** [review]
Have you check MaasClusterStatusStep is still working properly?

I think it would be broken by this change.

File: sunbeam-python/sunbeam/provider/maas/deployment.py [hunk 2]
```diff
@@ -39,6 +39,7 @@ class RoleTags(enum.Enum):
     COMPUTE = "compute"
     STORAGE = "storage"
     JUJU_CONTROLLER = "juju-controller"
+    CLUSTERD = "sunbeam-cluster"
```
[the remark this answers is not in the document]
Comment (hemanthnakkina) [reply sunbeam-python/sunbeam/provider/maas/deployment.py]: This is now renamed to INFRA and further used in code in RoleTags

File: sunbeam-python/sunbeam/provider/maas/deployment.py (hunk 1 above)
Comment (hemanthnakkina) [reply sunbeam-python/sunbeam/provider/maas/deployment.py]: model name changed to `openstack-infra`
property `infrastructue_model` renamed to `openstack_machines_model`

**Comment by hemanthnakkina** [comment]
> Have you check MaasClusterStatusStep is still working properly?
> 
> I think it would be broken by this change.

This is fixed now

**Review by gboutry — COMMENTED** [review]
MaasClusterStatusStep line 2140 will try to match the clusterd member to the controller model unit, but it will be now in the openstack infra model, so this can't work

**Comment by hemanthnakkina** [comment]
> MaasClusterStatusStep line 2140 will try to match the clusterd member to the controller model unit, but it will be now in the openstack infra model, so this can't work

Fixed in subsequent push

**Review by gboutry — APPROVED** [review]

**Pull request merged**